--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -435,9 +435,6 @@ install(DIRECTORY presets profiles DESTI
 
 if(UNIX AND NOT APPLE)
   install(FILES docs/melt.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 RENAME melt-${MLT_VERSION_MAJOR}.1)
-  install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink melt-${MLT_VERSION_MAJOR}.1 melt.1 \
-                                WORKING_DIRECTORY ${CMAKE_INSTALL_FULL_MANDIR}/man1)"
-  )
 endif()
 
 add_subdirectory(src)
--- a/src/melt/CMakeLists.txt
+++ b/src/melt/CMakeLists.txt
@@ -22,7 +22,4 @@ if(WIN32 OR APPLE)
   install(TARGETS melt RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 else()
   install(PROGRAMS "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/melt" DESTINATION ${CMAKE_INSTALL_BINDIR} RENAME melt-${MLT_VERSION_MAJOR})
-  install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink melt-${MLT_VERSION_MAJOR} melt \
-                                WORKING_DIRECTORY ${CMAKE_INSTALL_FULL_BINDIR})"
-  )
 endif()
