--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -324,7 +324,6 @@ endif()
 
 if(BUILD_TESTING)
   find_package(Qt${QT_MAJOR_VERSION} REQUIRED COMPONENTS Core Test)
-  find_package(Kwalify REQUIRED)
   enable_testing()
 endif()
 
--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -52,11 +51,3 @@ if(MOD_QT6)
 endif()
 
 
-file(GLOB YML_FILES "${CMAKE_SOURCE_DIR}/src/modules/*/*.yml")
-foreach(YML_FILE ${YML_FILES})
-  get_filename_component(FILE_NAME ${YML_FILE} NAME)
-  file(RELATIVE_PATH KWALIFY_TEST_NAME "${CMAKE_SOURCE_DIR}/src/modules" ${YML_FILE})
-  if(NOT FILE_NAME MATCHES "resolution_scale.yml")
-    add_test(NAME "kwalify:${KWALIFY_TEST_NAME}" COMMAND ${Kwalify_EXECUTABLE} -f "${CMAKE_SOURCE_DIR}/src/framework/metaschema.yaml" ${YML_FILE})
-  endif()
-endforeach()
