--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ project( libid3tag )
 
 include(CMakeDependentOption)
 
-set( TARGET libid3tag )
+set( TARGET id3tag )
 
 cmake_dependent_option( BUILD_SHARED "Build a shared library" Off "NOT WIN32" Off)
 
@@ -12,6 +12,7 @@ find_package(ZLIB REQUIRED)
 
 if( BUILD_SHARED )
    add_library( ${TARGET} SHARED )
+   set_target_properties( ${TARGET} PROPERTIES VERSION 0.3.0 SOVERSION 0 )
 else()
    add_library( ${TARGET} STATIC )
 endif()
