--- a/src/doc/Makefile
+++ b/src/doc/Makefile
@@ -79,12 +79,12 @@ SPHINXBUILDDIR      = ../../build/sphinx
 
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-sphinx : Makefile conf.py
+sphinx : doxygen.turd conf.py
 	# @$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(SPHINXBUILDDIR)" $(SPHINXOPTS) $(O)
 	$(SPHINXBUILD) -b html "$(SOURCEDIR)" "$(SPHINXBUILDDIR)" $(SPHINXOPTS)
 
-sphinxpdf : Makefile conf.py
+sphinxpdf : doxygen.turd conf.py
 	# @$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(SPHINXBUILDDIR)" $(SPHINXOPTS) $(O)
-	$(SPHINXBUILD) -b latex "$(SOURCEDIR)" "$(SPHINXBUILDDIR)/latex" $(SPHINXOPTS)
-	$(MAKE) -C $(SPHINXBUILDDIR)/latex all-pdf
+	$(SPHINXBUILD) -b latex "$(SOURCEDIR)" "$(SPHINXBUILDDIR)/../latex" $(SPHINXOPTS)
+	$(MAKE) -C $(SPHINXBUILDDIR)/../latex all-pdf
 # 	$(SPHINXBUILD) -b latexpdf "$(SOURCEDIR)" "$(SPHINXBUILDDIR)/latex" $(SPHINXOPTS)
