--- a/src/cmake/oiio_macros.cmake
+++ b/src/cmake/oiio_macros.cmake
@@ -70,7 +70,7 @@ macro (add_oiio_plugin)
             target_link_libraries (${_plugin_name} PUBLIC OpenImageIO
                                                    PRIVATE ${_plugin_LINK_LIBRARIES})
             set_target_properties (${_plugin_name} PROPERTIES PREFIX "" FOLDER "Plugins")
-            install_targets (${_plugin_name})
+            install(TARGETS ${_plugin_name} LIBRARY DESTINATION lib/openimageio)
         endif ()
     else ()
         message (STATUS "${ColorRed}Disabling ${_plugin_name} ${ColorReset}")
