--- a/zart/src/ImageConverter.cpp
+++ b/zart/src/ImageConverter.cpp
@@ -70,7 +70,7 @@ void ImageConverter::convert(const cv::M
 
   cv::Mat tmp(in->cols, in->rows, in->depth());
 
-  cvtColor(*in, tmp, (in->channels() == 1) ? CV_GRAY2RGB : CV_BGR2RGB);
+  cvtColor(*in, tmp, (in->channels() == 1) ? cv::COLOR_GRAY2RGB : cv::COLOR_BGR2RGB);
 
   const unsigned int w3 = 3 * tmp.cols;
   unsigned char * src = reinterpret_cast<unsigned char *>(tmp.ptr());
@@ -114,7 +114,7 @@ void ImageConverter::convert(const QImag
       dst += step;
     }
   }
-  cvtColor(**out, **out, CV_BGR2RGB);
+  cvtColor(**out, **out, cv::COLOR_BGR2RGB);
 }
 
 void ImageConverter::convert(const cimg_library::CImg<float> & in, QImage * out)
