--- a/cmake-proxies/CMakeLists.txt
+++ b/cmake-proxies/CMakeLists.txt
@@ -189,13 +190,6 @@ if( ${_OPT}has_networking )
       set ( curl_ssl "openssl" )
    endif ()
 
-   add_conan_lib(
-      ThreadPool
-      threadpool/20140926
-      REQUIRED
-      ALWAYS_ALLOW_CONAN_FALLBACK
-   )
-
    add_conan_lib(
       CURL
       libcurl/7.75.0
@@ -227,6 +221,7 @@ if( ${_OPT}has_sentry_reporting )
       RapidJSON
       rapidjson/1.1.0
       REQUIRED
+      PKG_CONFIG "RapidJSON"
    )
 endif()
 
--- a/libraries/lib-network-manager/CMakeLists.txt
+++ b/libraries/lib-network-manager/CMakeLists.txt
@@ -32,7 +32,6 @@ set( SOURCES
 
 set ( LIBRARIES PRIVATE
     CURL::libcurl
-    ThreadPool::ThreadPool
     lib-string-utils-interface
     lib-exceptions-interface
     lib-utility-interface
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -760,7 +760,9 @@ list( APPEND SOURCES
       tracks/playabletrack/notetrack/ui/NoteTrackControls.cpp
       tracks/playabletrack/notetrack/ui/NoteTrackControls.h
       tracks/playabletrack/notetrack/ui/NoteTrackMenuItems.cpp
+      $<$<BOOL:${USE_MIDI}>:
       tracks/playabletrack/notetrack/ui/NoteTrackShifter.cpp
+      >
       tracks/playabletrack/notetrack/ui/NoteTrackSliderHandles.cpp
       tracks/playabletrack/notetrack/ui/NoteTrackSliderHandles.h
       tracks/playabletrack/notetrack/ui/NoteTrackVRulerControls.cpp
