From 4cea4f4cc054270088b64dd1f532794e57a21afc Mon Sep 17 00:00:00 2001
From: John Arbuckle <programmingkidx@gmail.com>
Date: Fri, 11 Mar 2022 17:56:12 -0500
Subject: [PATCH] Add check for correct SIP version.

---
 buildtools/wxpysip.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/buildtools/wxpysip.py b/buildtools/wxpysip.py
index c58102b8..9a61d2a6 100644
--- a/buildtools/wxpysip.py
+++ b/buildtools/wxpysip.py
@@ -16,6 +16,7 @@
 #       run a subprocess, and to also add a little missing sip 4 functionality
 #       that we were depending on with the old SIP.
 import os
+import sys
 
 from sipbuild.code_generator import (set_globals, parse, generateCode,
         generateExtracts, generateAPI, generateXML, generateTypeHints)
@@ -23,6 +24,10 @@ from sipbuild.exceptions import handle_exception, UserException
 from sipbuild.module import resolve_abi_version
 from sipbuild.version import SIP_VERSION, SIP_VERSION_STR
 
+print("SIP version: " + str(SIP_VERSION_STR))
+if SIP_VERSION_STR != "5.5.0":
+    print("Error: only SIP version 5.5.0 is supported.")
+    sys.exit(-1)
 
 def sip_runner(
     specification,              # the name of the specification file [default stdin]
-- 
2.32.0 (Apple Git-132)

