--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ project( libmad )
 
 include(CMakeDependentOption)
 
-set( TARGET libmad )
+set( TARGET mad )
 
 cmake_dependent_option( BUILD_SHARED "Build a shared library" Off "NOT WIN32" Off)
 option( OPT_ACCURACY "Favour accuracy" On)
@@ -13,6 +13,7 @@ option( OPT_SSO "Enable a fast subband s
 
 if( BUILD_SHARED )
    add_library( ${TARGET} SHARED )
+   set_target_properties( ${TARGET} PROPERTIES VERSION 0.2.1 SOVERSION 0 )
 else()
    add_library( ${TARGET} STATIC )
 endif()
