--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,7 +147,7 @@ IF ( CMAKE_HOST_UNIX )
              ${MPI_LIBRARIES} )
   ENDIF ( MPI_FOUND )
   LIST ( APPEND Field3D_Libraries_Shared
-    Iex Half IlmThread Imath
+    Iex IlmThread Imath
     pthread dl z )
   SET ( Field3D_DSO_Libraries ${Field3D_Libraries_Shared} )
   SET ( Field3D_BIN_Libraries Field3D ${Field3D_Libraries_Shared}
--- a/export/SpiMathLib.h
+++ b/export/SpiMathLib.h
@@ -39,7 +39,7 @@
 #define _INCLUDED_Field3D_SpiMathLib_H_
 
 #include <half.h>
-#include <ImathHalfLimits.h>
+#include <halfLimits.h>
 
 #include <ImathBox.h>
 #include <ImathBoxAlgo.h>
--- a/export/StdMathLib.h
+++ b/export/StdMathLib.h
@@ -41,7 +41,7 @@
 #include <ImathBox.h> 
 #include <ImathBoxAlgo.h>
 #include <ImathColor.h>
-#include <ImathHalfLimits.h>
+#include <halfLimits.h>
 #include <ImathMatrix.h>
 #include <ImathMatrixAlgo.h>
 #include <ImathPlane.h>
