--- a/Makefile.in
+++ b/Makefile.in
@@ -4063,6 +4063,7 @@ COND_USE_GUI_1_ALL_GUI_HEADERS =  \
 	wx/generic/creddlgg.h \
 	wx/creddlg.h \
 	wx/bmpbndl.h \
+	wx/filedlgcustomize.h \
 	$(LOWLEVEL_HDR) \
 	$(GUI_CORE_HEADERS) \
 	wx/mediactrl.h \
diff --git a/build/bakefiles/files.bkl b/build/bakefiles/files.bkl
index 0b16b7d6326b..5007c6ab97b8 100644
--- a/build/bakefiles/files.bkl
+++ b/build/bakefiles/files.bkl
@@ -1310,6 +1310,7 @@ IMPORTANT: please read docs/tech/tn0016.txt before modifying this file!
     wx/generic/creddlgg.h
     wx/creddlg.h
     wx/bmpbndl.h
+    wx/filedlgcustomize.h
 </set>
 
 <!-- ====================================================================== -->
