--- a/configure.ac
+++ b/configure.ac
@@ -8582,15 +8582,17 @@
     AC_MSG_CHECKING([whether catch.hpp file exists])
     if ! test -f "$ac_confdir/3rdparty/catch/include/catch.hpp" ; then
         AC_MSG_RESULT([no])
+    PKG_CHECK_MODULES(CATCH, [catch2], [CFLAGS="$CATCH_CFLAGS $CFLAGS"], [
         AC_MSG_ERROR([
     CATCH (C++ Automated Test Cases in Headers) is required, the required file
     $ac_confdir/3rdparty/catch/include/catch.hpp couldn't be found.
 
     You might need to run
 
         git submodule update --init 3rdparty/catch
 
     to fix this.])
+    ])
     else
         AC_MSG_RESULT([yes])
     fi
