--- a/build/cmake/functions.cmake
+++ b/build/cmake/functions.cmake
@@ -1057,6 +1057,7 @@ endfunction()
 # The default is ON if not third parameter is specified
 function(wx_option name desc)
     cmake_parse_arguments(OPTION "" "" "STRINGS" ${ARGN})
+    set(OPTION_STRINGS OFF)
     if(ARGC EQUAL 2)
         set(default ON)
     else()
--- a/src/html/helpctrl.cpp
+++ b/src/html/helpctrl.cpp
@@ -30,7 +30,7 @@
 
 #if wxUSE_LIBMSPACK
 #include "wx/html/forcelnk.h"
-FORCE_LINK(wxhtml_chm_support)
+FORCE_LINK_ME(wxhtml_chm_support)
 #endif
 
 wxIMPLEMENT_DYNAMIC_CLASS(wxHtmlHelpController, wxHelpControllerBase);
