--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ include(build/cmake/policies.cmake NO_POLICY_SCOPE)
 # Initialize variables for quick access to wx root dir in sub dirs
 set(wxSOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set(wxBINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
-set(wxOUTPUT_DIR ${wxBINARY_DIR}/lib)
+set(wxOUTPUT_DIR ${wxBINARY_DIR}/${CMAKE_INSTALL_LIBDIR})
 
 # parse the version number from wx/version.h and include in wxMAJOR_VERSION and wxMINOR_VERSION
 file(READ "${wxSOURCE_DIR}/include/wx/version.h" WX_VERSION_H_CONTENTS)
--- a/build/cmake/config.cmake
+++ b/build/cmake/config.cmake
@@ -110,7 +110,7 @@ function(wx_write_config)
     set(prefix ${CMAKE_INSTALL_PREFIX})
     set(exec_prefix "\${prefix}")
     set(includedir "\${prefix}/include")
-    set(libdir "\${exec_prefix}/lib")
+    set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
     set(bindir "\${exec_prefix}/bin")
 
     if(wxBUILD_MONOLITHIC)
--- a/build/cmake/functions.cmake
+++ b/build/cmake/functions.cmake
@@ -456,8 +456,8 @@ macro(wx_add_library name)
         endif()
         wx_install(TARGETS ${name}
             EXPORT wxWidgetsTargets
-            LIBRARY DESTINATION "lib${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}"
-            ARCHIVE DESTINATION "lib${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}"
+            LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}"
+            ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}"
             RUNTIME DESTINATION "${runtime_dir}${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}"
             BUNDLE DESTINATION Applications/wxWidgets
             )
@@ -581,7 +581,7 @@ function(wx_set_builtin_target_properties target_name)
 
     wx_set_common_target_properties(${target_name} DEFAULT_WARNINGS)
     if(NOT wxBUILD_SHARED)
-        wx_install(TARGETS ${name} EXPORT wxWidgetsTargets ARCHIVE DESTINATION "lib${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}")
+        wx_install(TARGETS ${name} EXPORT wxWidgetsTargets ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}${GEN_EXPR_DIR}${wxPLATFORM_LIB_DIR}")
     endif()
 endfunction()
 
--- a/build/cmake/install.cmake
+++ b/build/cmake/install.cmake
@@ -25,15 +25,15 @@ endif()
 if(WIN32_MSVC_NAMING)
     install(
         DIRECTORY "${wxSETUP_HEADER_PATH}"
-        DESTINATION "lib/${wxPLATFORM_LIB_DIR}")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/${wxPLATFORM_LIB_DIR}")
 else()
     install(
         DIRECTORY "${wxSETUP_HEADER_PATH}"
-        DESTINATION "lib/wx/include")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/wx/include")
 
     install(
         FILES "${wxOUTPUT_DIR}/wx/config/${wxBUILD_FILE_ID}"
-        DESTINATION "lib/wx/config"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/wx/config"
         PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
                     GROUP_EXECUTE GROUP_READ
                     WORLD_EXECUTE WORLD_READ
@@ -48,7 +48,7 @@ else()
     )
 endif()
 
-install(EXPORT wxWidgetsTargets NAMESPACE wx:: DESTINATION "lib/cmake/wxWidgets/${wxPLATFORM_LIB_DIR}")
+install(EXPORT wxWidgetsTargets NAMESPACE wx:: DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/wxWidgets/${wxPLATFORM_LIB_DIR}")
 
 # find_package config file
 include(CMakePackageConfigHelpers)
@@ -71,11 +71,11 @@ write_basic_package_version_file(
 configure_package_config_file(
     "${wxSOURCE_DIR}/build/cmake/wxWidgetsConfig.cmake.in"
     "${projectConfig}"
-    INSTALL_DESTINATION "lib/cmake/wxWidgets"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/wxWidgets"
 )
 install(
     FILES "${projectConfig}" "${versionConfig}"
-    DESTINATION "lib/cmake/wxWidgets"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/wxWidgets"
 )
 
 # uninstall target
--- a/build/cmake/lib/webview/CMakeLists.txt
+++ b/build/cmake/lib/webview/CMakeLists.txt
@@ -150,7 +150,7 @@ if(WXGTK AND wxUSE_WEBVIEW_WEBKIT2)
         ${WEBKIT2_LIBRARIES}
         )
 
-    wx_install(TARGETS wxwebkit2_ext LIBRARY DESTINATION "lib/wx/${WX_WEB_EXT_VERSION}/web-extensions")
+    wx_install(TARGETS wxwebkit2_ext LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/wx/${WX_WEB_EXT_VERSION}/web-extensions")
 
     wx_add_dependencies(wxwebview wxwebkit2_ext)
 endif()
