--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -314,6 +314,8 @@ if(${OPENCOLORIO_FOUND})
   target_link_libraries(IO
     PUBLIC
       ${OPENCOLORIO_LIBRARIES}
+      "dl"
+      "GLU"
   )
 else()
   message(STATUS "  Not adding OpenColorIO nodes")
@@ -382,7 +383,7 @@ if(${SEEXPR2_FOUND})
   )
   target_link_libraries(IO
     PUBLIC
-      ${SEEXPR2_LIBRARIES}
+      ${SEEXPR2_LIBRARY}
   )
 else()
   message(STATUS "  Not adding SeExpr nodes")
--- a/cmake/Modules/FindOpenImageIO.cmake
+++ b/cmake/Modules/FindOpenImageIO.cmake
@@ -59,6 +59,10 @@ find_library ( OPENIMAGEIO_LIBRARY
                NAMES OpenImageIO${OIIO_LIBNAME_SUFFIX}
                HINTS ${OPENIMAGEIO_ROOT_DIR}
                PATH_SUFFIXES lib64 lib )
+find_library ( OPENIMAGEIO_UTIL_LIBRARY
+               NAMES OpenImageIO_Util${OIIO_LIBNAME_SUFFIX}
+               HINTS ${OPENIMAGEIO_ROOT_DIR}
+               PATH_SUFFIXES lib64 lib )
 find_path ( OPENIMAGEIO_INCLUDE_DIR
             NAMES OpenImageIO/imageio.h
             HINTS ${OPENIMAGEIO_ROOT_DIR} )
@@ -97,6 +101,12 @@ set (OPENIMAGEIO_FOUND ${OpenImageIO_FOU
 if (OpenImageIO_FOUND)
     set (OPENIMAGEIO_INCLUDES ${OPENIMAGEIO_INCLUDE_DIR})
     set (OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_LIBRARY})
+    if (${OPENIMAGEIO_VERSION} VERSION_GREATER_EQUAL "2.3" AND NOT TARGET OpenImageIO::OpenImageIO_Util)
+        add_library(OpenImageIO::OpenImageIO_Util UNKNOWN IMPORTED)
+        set_target_properties(OpenImageIO::OpenImageIO_Util PROPERTIES
+            IMPORTED_LOCATION "${OPENIMAGEIO_UTIL_LIBRARY}")
+        list (APPEND OPENIMAGEIO_LIBRARIES ${OPENIMAGEIO_UTIL_LIBRARY})
+    endif ()
     get_filename_component (OPENIMAGEIO_LIBRARY_DIRS "${OPENIMAGEIO_LIBRARY}" DIRECTORY)
     if (NOT OpenImageIO_FIND_QUIETLY)
         message ( STATUS "OpenImageIO includes     = ${OPENIMAGEIO_INCLUDE_DIR}" )
