--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -314,6 +314,7 @@ if(${OPENCOLORIO_FOUND})
   target_link_libraries(IO
     PUBLIC
       ${OPENCOLORIO_LIBRARIES}
+      "GLU"
   )
 else()
   message(STATUS "  Not adding OpenColorIO nodes")
@@ -382,7 +383,7 @@ if(${SEEXPR2_FOUND})
   )
   target_link_libraries(IO
     PUBLIC
-      ${SEEXPR2_LIBRARIES}
+      ${SEEXPR2_LIBRARY}
   )
 else()
   message(STATUS "  Not adding SeExpr nodes")
