--- a/EXR/WriteEXR.cpp
+++ b/EXR/WriteEXR.cpp
@@ -32,6 +32,11 @@ GCC_DIAG_OFF(deprecated)
 #include <ImfArray.h>
 #include <ImfOutputFile.h>
 #include <half.h>
+#if OPENEXR_VERSION_MAJOR >= 3
+#include <ImfHeader.h>
+#include <ImfFrameBuffer.h>
+#include <ImathBox.h>
+#endif
 GCC_DIAG_ON(deprecated)
 
 #include "GenericOCIO.h"
--- a/EXR/ReadEXR.cpp
+++ b/EXR/ReadEXR.cpp
@@ -43,6 +43,11 @@ GCC_DIAG_OFF(deprecated)
 #include <ImfChannelList.h>
 #include <ImfInputFile.h>
 #include <IlmThreadPool.h>
+#if OPENEXR_VERSION_MAJOR >= 3
+#include <ImfHeader.h>
+#include <ImfFrameBuffer.h>
+#include <ImathBox.h>
+#endif
 GCC_DIAG_ON(deprecated)
 
 #ifdef OFX_IO_MT_EXR
